build-image-src/Dockerfile-dotnet7

```diff
@@ -0,0 +1,89 @@
+ARG IMAGE_ARCH
+FROM public.ecr.aws/sam/lambda/dotnet7:latest-$IMAGE_ARCH
```

Comment:
@mndeveci, the Lambda team is still blocking the release of the dotnet7 image we want to base off of. Would it be okay for now to just put that code here for now and not base of this image? Then when they release their image, we can change this back here to base off their image.

The code for their base image is here (it's just not released yet) https://code.amazon.com/reviews/CR-77676031/revisions/2#/diff 

@smhmhmd FYI

Reply:
Sure, can we test it locally (by building dotnet7 image on your laptop and try to build this image as well)?

Reply:
@mndeveci, sure does this work for tests? Anything to add? 

Test 1. Create my own runtime image locally using the dockerfile from the Lambda CR, make sure I can use it as a runtime image to run a .NET 7 app in Lambda in a container (runtime container, not build container).

Test 2. Copy their code into the draft PR sam-build-images .NET 7 build image (replacing the `FROM public.ecr.aws/sam/lambda/dotnet7:latest-$IMAGE_ARCH`) and build that locally. Then make sure I can build and run a .NET 7 and .NET 7 Native AOT custom runtime image in Lambda on provided AL2